packages/plugin-core/src/test/suite-integ/EngineNoteProvider.test.ts

```diff
@@ -1,169 +1,317 @@
-                  } as Error);
+      },
+      () => {
+        testAsyncWithCallback(
+          "THEN the tree data is correct",
+          {},
+          async (done) => {
+            const mockEvents = new MockEngineEvents();
+            const provider = new EngineNoteProvider(mockEvents);
+
+            const props = await (provider.getChildren() as Promise<
+              NoteProps[]
+            >);
+
+            // 3 Vaults hence 3 root nodes
+            expect(props.length === 3);
+
+            // Also check some children:
+
+            // tracking errors to avoid calling done twice.
+            let error: DendronError | undefined;
+            props.forEach((props) => {
+              switch (props.vault.fsPath) {
+                case "vault1": {
+                  if (
+                    props.children.length !== 1 ||
+                    props.children[0] !== "foo"
+                  ) {
+                    error = new DendronError({
+                      message: "Note children in vault1 incorrect!",
+                    });
+                    done(error);
+                  }
+                  break;
                 }
-                break;
-              }
-              case "vault2": {
-                if (
-                  props.children.length !== 1 ||
-                  props.children[0] !== "bar"
-                ) {
-                  done({
-                    message: "Note children in vault2 incorrect!",
-                  } as Error);
+                case "vault2": {
+                  if (
+                    props.children.length !== 1 ||
+                    props.children[0] !== "bar"
+                  ) {
+                    error = new DendronError({
+                      message: "Note children in vault2 incorrect!",
+                    });
+                    done(error);
+                  }
+                  break;
                 }
-                break;
-              }
-              case "vault3": {
-                if (props.children.length !== 0) {
-                  done({
-                    message: "Note children in vault3 incorrect!",
-                  } as Error);
+                case "vault3": {
+                  if (props.children.length !== 0) {
+                    error = new DendronError({
+                      message: "Note children in vault3 incorrect!",
+                    });
+                    done(error);
+                  }
+                  break;
+                }
+                default: {
+                  error = new DendronError({
+                    message: "Note with unexpected vault found!",
+                  });
+                  done(error);
                 }
-                break;
-              }
-              default: {
-                done({ message: "Note with unexpected vault found!" } as Error);
               }
+            });
+
+            if (!error) done();
```

Comment:
`done` was being called twice when it failed which made the error message hard to read.